notification/src/lambda.js

```diff
@@ -0,0 +1,29 @@
+const ctp = require('./utils/ctp')
+const handler = require('./handler/notification/notification.handler')
+const config = require('./config/config')()
+const logger = require('./utils/logger').getLogger()
+
+const setup = require('./config/init/ensure-interface-interaction-custom-type')
+
+const ctpClient = ctp.get(config)
+
+let initialised = false
+
+// TODO: add JSON schema validation:
+// https://github.com/commercetools/commercetools-adyen-integration/issues/9
+exports.handler = async function (event) {
+  try {
+    if (!initialised) {
+      await setup.ensureInterfaceInteractionCustomType(ctpClient)
```

Comment:
So here as opposite to extensions it makes sense to do initialisation?

Reply:
Yes, in this case it would work because the lambda will be invoked before the custom types are needed. Something I would like to discuss either here or in a separate issue is having a potential environment variable flag that disables this behaviour as in our case we do not want this to be created by the lambda, we want it managed by terraform.

Reply:
Please create another ticket for it. In general we wanted to keep mandatory setups close to the code as in the future if we have to add new fields to existing custom type it would automatically update it. Of course permance-wise it would be better no to check on every cold start but on deployment only.

Reply:
Alright, yea that's understandable, but for people like us who are trying to keep the infrastructure setup from the code, this doesn't really work particularly well. I'll create an issue detailing the use case, and then work on getting the code for this done, it shouldn't be to much work hopefully.